<a href="https://colab.research.google.com/github/mridu-pant/ML-Basics/blob/main/json_files_and_weather_api.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import json
import urllib.request as url

In [3]:
path = "https://data.covid19india.org/states_daily.json"
response = url.urlopen(path)

In [4]:
data = json.load(response)

In [5]:
type(data)

dict

In [6]:
data.keys()

dict_keys(['states_daily'])

In [7]:
states = data["states_daily"]

In [8]:
len(states)

1563

In [9]:
len(states)/3

521.0

**Conclusions**
1.   Given data is stored as dict
2.   Key is 'states_daily'
3.   For each day, there is a total of 3 different types of data given: Confirmed cases, Deceased, and Recovered.
4. Number of days = 521



In [10]:
import pandas as pd

In [12]:
df = pd.DataFrame(states);

In [14]:
df.head()

an ap ar as br ch ct       date     dateymd dd  ... sk     status tg tn tr  \
0  0  1  0  0  0  0  0  14-Mar-20  2020-03-14  0  ...  0  Confirmed  1  1  0   
1  0  0  0  0  0  0  0  14-Mar-20  2020-03-14  0  ...  0  Recovered  0  0  0   
2  0  0  0  0  0  0  0  14-Mar-20  2020-03-14  0  ...  0   Deceased  0  0  0   
3  0  0  0  0  0  0  0  15-Mar-20  2020-03-15  0  ...  0  Confirmed  2  0  0   
4  0  0  0  0  0  0  0  15-Mar-20  2020-03-15  0  ...  0  Recovered  1  0  0   

   tt un  up ut wb  
0  81  0  12  0  0  
1   9  0   4  0  0  
2   2  0   0  0  0  
3  27  0   1  0  0  
4   4  0   0  0  0  

[5 rows x 42 columns]

In [17]:
API_key = "23313c826d7615f0c781a86adf142256"
city_name = "Delhi"
path = f"https://api.openweathermap.org/data/2.5/weather?q={city_name}&appid={API_key}"
request = url.Request(path, method = 'GET')
response = url.urlopen(request)
data = json.load(response)

In [19]:
len(data)

13

In [20]:
data.keys()

dict_keys(['coord', 'weather', 'base', 'main', 'visibility', 'wind', 'clouds', 'dt', 'sys', 'timezone', 'id', 'name', 'cod'])

In [23]:
data

{'base': 'stations',
 'clouds': {'all': 40},
 'cod': 200,
 'coord': {'lat': 28.6667, 'lon': 77.2167},
 'dt': 1655906675,
 'id': 1273294,
 'main': {'feels_like': 308.67,
  'humidity': 40,
  'pressure': 1004,
  'temp': 307.2,
  'temp_max': 307.2,
  'temp_min': 307.2},
 'name': 'Delhi',
 'sys': {'country': 'IN',
  'id': 9165,
  'sunrise': 1655855635,
  'sunset': 1655905927,
  'type': 1},
 'timezone': 19800,
 'visibility': 4000,
 'weather': [{'description': 'haze',
   'icon': '50n',
   'id': 721,
   'main': 'Haze'}],
 'wind': {'deg': 340, 'speed': 4.12}}